models/generic_transaction.go

```diff
@@ -40,6 +40,23 @@ type TransactionWithBatchDetails struct {
 	BatchTime   *Timestamp
 }
 
+func FindOldestTransactionTime(array GenericTransactionArray) *Timestamp {
```

Comment:
I'm not sure it makes sense to move this into `utils`, that would make this (though I didn't look very carefully) the only function in `utils` which imports `models`, and introduce an import cycle where `utils` imports `models` which imports `utils` again.

An alternative which I considered but didn't do is to make it a method on the `GenericTransactionArray` interface and then implement it for each of the three types. I'm happy to do that if you think it would be better!

I'm adding a few unit tests now.

Reply:
Ahh sorry, I noticed your reply just now. You're right we'd have a cycle if we moved `FindOldestTransactionTime` to `utils`. I moved it to a new file in `executor` package instead, because we try to keep `models` package thin and only put methods that directly relate to the data structures in there (like encoders/decoders).